plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/compare/changedpanel/ChangedPanelPresenter.java

```diff
@@ -0,0 +1,132 @@
+    private final GitLocalizationConstant locale;
+
+    private Map<String, Status> changedFiles;
+    private CallBack            callBack;
+    private boolean             treeViewEnabled;
+
+    @Inject
+    public ChangedPanelPresenter(GitLocalizationConstant locale,
+                                 ChangedPanelView view,
+                                 AppContext appContext,
+                                 NotificationManager notificationManager,
+                                 ComparePresenter comparePresenter) {
+        this.locale = locale;
+        this.view = view;
+        this.appContext = appContext;
+        this.notificationManager = notificationManager;
+        this.comparePresenter = comparePresenter;
+        this.view.setDelegate(this);
+        this.treeViewEnabled = true;
+    }
+
+    /**
+     * Show panel with changed files. If empty map with changed files is received, all buttons would be disabled.
+     *
+     * @param changedFiles
+     *         Map with files and their status
+     */
+    public void show(Map<String, Status> changedFiles, @Nullable CallBack callBack) {
+        this.changedFiles = changedFiles;
+        this.callBack = callBack;
+        if (changedFiles.isEmpty()) {
+            view.setTextToChangeViewModeButton(locale.changeListRowListViewButtonText());
+            view.setEnabledChangeViewModeButton(false);
+            view.setEnableExpandCollapseButtons(false);
+            view.clearNodeStorage();
+        } else {
+            view.setEnabledChangeViewModeButton(true);
+            view.setEnableExpandCollapseButtons(treeViewEnabled);
+            viewChangedFiles();
+        }
+    }
+
+    public ChangedPanelView getView() {
+        return view;
+    }
+
+    @Override
+    public void onFileNodeDoubleClicked(String path, final Status status) {
+        appContext.getRootProject()
+                  .getFile(path)
+                  .then(file -> {
+                      if (file.isPresent()) {
+                          comparePresenter.showCompareWithLatest(file.get(), status, "HEAD");
+                      }
+                  })
+                  .catchError(error -> {
+                      notificationManager.notify(error.getMessage(), FAIL, NOT_EMERGE_MODE);
+                  });
+    }
+
+    @Override
+    public void onChangeViewModeButtonClicked() {
+        treeViewEnabled = !treeViewEnabled;
+        viewChangedFiles();
+        view.setEnableExpandCollapseButtons(treeViewEnabled);
+    }
+
+    @Override
+    public void onExpandButtonClicked() {
+        view.expandAllDirectories();
+    }
+
+    @Override
+    public void onCollapseButtonClicked() {
+        view.collapseAllDirectories();
+    }
+
+    @Override
+    public void onNodeSelected(Node node) {
+        if (callBack != null) {
+            callBack.onNodeSelected(node);
+        }
+    }
+
+    private void viewChangedFiles() {
+        if (treeViewEnabled) {
```

Comment:
Replace `treeViewEnabled` with enum with two values `TREE` and `LIST`. It would be better to understand the logic